CryoGrid3_xice_mpi.m

```diff
@@ -398,21 +420,54 @@
+                                water_fluxes = calculateLateralWaterDarcyFluxes( T, PACKAGE_waterExchange_j, GRID, PARA, j, water_fluxes);  % matrix containing all fluxes in [m/s] scaled to row index
                             end
                         end
-                        waterflux = nansum( water_fluxes.*PARA.technical.syncTimeStep.*24.*3600 ); % in m
+                        
+                        % Calculate possible boundary fluxes
+                        [ boundary_water_flux ] = calculateLateralWaterBoundaryFluxes(PARA, GRID, T);
+                        
+                        % Check for water availability and set real water fluxes
+                        [ water_fluxes_worker, boundary_water_flux ] = calculateLateralWaterAvailable( PARA,GRID, wc, water_fluxes,boundary_water_flux );
+                        water_fluxes_gather=zeros(numlabs,numlabs,numlabs);
+                        water_fluxes_gather(:,:,labindex)=water_fluxes_worker;
+                        
+                        % Send real fluxes all around
+                        for j=1:number_of_realizations
+                            if j~=index
+                                labSend( water_fluxes_worker, j, 5);
+                            end
+                        end
+                        
+                        for j=1:number_of_realizations
+                            if j~=index
+                                water_fluxes_worker_j = labReceive(j, 5);
+                                water_fluxes_gather(:,:,j)=water_fluxes_worker_j;
+                            end
+                        end
+                        
+                        water_fluxes=nansum(water_fluxes_gather,3);
+                        waterflux=nansum(water_fluxes(labindex,:))+boundary_water_flux;
+                        
                         % apply lateral water flux directly (as bulk subsurface flux)
                         [wc, excess_water, lacking_water] = bucketScheme(T, wc, zeros( size(wc) ), GRID, PARA, waterflux);
-                        assert( lacking_water==0, 'CryoGrid3 - lateral exchange - lacking water>0');    % there should be no lacking water as this was checked for
-                        if excess_water>0                                                               % in case of s
-                            GRID.lake.residualWater = GRID.lake.residualWater + excess_water;         % for now: store the excess water, later: treat it according to BC for surface water fluxes
+                        assert( lacking_water < 1e-9, 'CryoGrid3 - lateral exchange - lacking water>0');    % there should be no lacking water as this was checked for
+                        
+                        % Store and display
+                        BALANCE.water.dr_water_fluxes_out=BALANCE.water.dr_water_fluxes_out+water_fluxes.*1000;
```

Comment:
I guess it depends how much you are interested in water fluxes. I think it can be cool to easily have access to the contribution of each worker to understand how your network works. Is it an issue to store a matrix rather than a vector ?